Add Fallout

televandalist opened the pull request:
Good grief, that was hell and a half.

- Performed full "humongous" install using [Fallout (USA) (Rerelease) (1999-11-10)](http://redump.org/disc/63721/), which is v1.1.

Sound Config:
- Card: Sound Blaster 16/AWE32
- Port: 220
- IRQ: 7
- DMA: 5

Edited the file paths of these lines in `FALLOUT.CFG`:
- music_path1=C:\data\sound\music\
- music_path2=C:\data\sound\music\
- critter_dat=C:\critter.dat
- critter_patches=C:\data
- cmaster_dat=C:\master.dat
- master_patches=C:\data

DOSC mounts the CD, auto-starts, and contains `FALLOUT.CFG` and `SOUND.EXE`.

Variants:
- Added a variant of v1.1 that's patched to v1.2 and then v1.3.5 [bugfixes by TeamX](https://www.nma-fallout.com/resources/fallout-v-1-3-5-patch-by-teamx-unofficial.53/)
  - this uses an identical DOSC, just a different name to match
- Added a variant for the vanilla v1.1 installation media

PureDOS commented:
Can you try to launch the game, select NEW GAME, click TAKE on the first character, skip the opening movie, then in the game click on PIP in the lower right, then click on ARCHIVES, then select "Overseer Briefing" and check if the game crashes or not?

televandalist commented:
> Can you try to launch the game, select NEW GAME, click TAKE on the first character, skip the opening movie, then in the game click on PIP in the lower right, then click on ARCHIVES, then select "Overseer Briefing" and check if the game crashes or not?

Well, that's pretty specific 😆 

It crashed for me on both the main DOSZ and bugfix variant. Seems like something that would be the result of a bad file path in the configs, but I'm pretty sure these are correct:
```
music_path1=C:\data\sound\music\
music_path2=C:\data\sound\music\
critter_dat=C:\critter.dat
critter_patches=C:\data
master_dat=C:\master.dat
master_patches=C:\data
```

I think all I did was remove the `INTERPLAY` directory from everything.

televandalist commented:
Never mind. Fixed it. I had `CD:\` instead of `CD\` in the `DOS.YML` for `run_mount`.

tomojira commented:
Trying a test/review of this! Here's what I found.

I think `UNINST.LOG` can be removed. Without it, it's possible to build the DOSZ using DoDAT without going through the installation process.

I think the CD image can be left out too. The humongous installation includes everything, and paths are edited in the config file, so the CD doesn't seem needed to run the game. `run_mount` can also be removed.

`FALLOUT.CFG` can be included in the DOSZ since the game can't run without it.
It's already included in the patch variant DOSZ.

The Video Archives crash can be fixed by increasing the memory size. I used `mem_size: 65536`.

`run_input` can be updated to `(WAITMODECHANGE)`.

A CPU setting can be included for consistency. `cpu_max_year: 1997` seems to be fine.

Installation variant works fine using DoDAT. Everything else seems good!

televandalist commented:
Yeah, the older PRs still need some tightening up 😆 I've been meaning to come back to these.

Did you try watching the videos from the archives menu? Would really like to figure out how to get it to not crash.

tomojira commented:
Yep, I mentioned it in that post, but it's because the default 16 MB memory size is too low. `mem_size: 65536` for example fixes it.

televandalist commented:
That's what I get for skimming posts 😭 

I'll try the tweaks mentioned and update it

televandalist commented:
Confirmed all of the above and made the following changes:
- Added `mem_size: 65536` and `cpu_max_year: 1997` to `DOS.YML`
- Removed the `run_mount: FALLOUT.CHD` line in `DOS.YML`
- Changed `(WAIT:1000)` to `(WAITMODECHANGE)` in `DOS.YML`
- Removed the CHD from the DOSZ files
- Moved the installation variant from `PureDOSDAT-Variants.xml` to `PureDOSDAT_Variants_Install.xml`
- Removed `UNINST.LOG` from the DOSZ files
- Added `FALLOUT.CFG` to the DOSZ files

televandalist commented:
> Can you try to launch the game, select NEW GAME, click TAKE on the first character, skip the opening movie, then in the game click on PIP in the lower right, then click on ARCHIVES, then select "Overseer Briefing" and check if the game crashes or not?

The crash is fixed now
